gr/include/gnuradio/tag.h

```diff
@@ -44,4 +85,60 @@ class tag_t
     pmtf::map _map;
 };
 
+// clang-format off
+constexpr const char* GR_TAG_PREFIX = "gr:";
+
+template <typename PMT_TYPE>
+    requires(pmtf::is_pmt_derived<PMT_TYPE>::value)
+class default_tag
+{ // TODO: consider replacing with a compile-time constexpr fixed-length string
+    const std::string _key;
+    const std::string _keyShort;
+    std::string _unit;
+    const std::string _description;
+
+public:
+    using value_type = PMT_TYPE::value_type;
+    default_tag(const std::string_view& key,
+                const std::string_view& unit = "",
+                const std::string_view& description = "")
+        : _key(std::string{ GR_TAG_PREFIX }.append(key)),
+          _keyShort(key),
+          _unit(unit),
+          _description(description)
+    {
+    }
+    default_tag() = delete;
+
+    [[nodiscard]] const std::string& key() const noexcept { return _key; }
+    [[nodiscard]] const std::string& shortKey() const noexcept { return _keyShort; }
+    [[nodiscard]] const std::string& unit() const noexcept { return _unit; }
+    [[nodiscard]] const std::string& description() const noexcept { return _description; }
+
+    [[nodiscard]] operator const std::string&() const noexcept { return _key; }
+
+    template <typename T>
+        requires std::is_same_v<value_type, T>
+    [[nodiscard]] std::pair<std::string, pmtf::pmt>
+    operator()(const T& newValue) const noexcept
+    {
+        return { _key, static_cast<pmtf::pmt>(PMT_TYPE(newValue)) };
+    }
+};
+
+namespace tag { // definition of default tags and names
+//TODO: change 'inline static const' to 'inline constexpr' once PMTF supports constexpr
+inline static const default_tag<pmtf::scalar<float>>    SAMPLE_RATE    = { "sample_rate", "Hz", "signal sample rate"};
+inline static const default_tag<pmtf::scalar<float>>    SIGNAL_RATE    = SAMPLE_RATE;
+inline static const default_tag<pmtf::string>           SIGNAL_NAME    = { "signal_name", "", "signal name"};
+inline static const default_tag<pmtf::string>           SIGNAL_UNIT    = { "signal_unit", "", "signal's physical SI unit"};
+inline static const default_tag<pmtf::scalar<float>>    SIGNAL_MIN     = { "signal_min", "a.u.", "signal physical max. (e.g. DAQ) limit"};
+inline static const default_tag<pmtf::scalar<float>>    SIGNAL_MAX     = { "signal_max", "a.u.", "signal physical max. (e.g. DAQ) limit"};
+inline static const default_tag<pmtf::string>           TRIGGER_NAME   = { "trigger_name"};
+inline static const default_tag<pmtf::scalar<uint64_t>> TRIGGER_TIME   = { "trigger_time", "ns", "UTC-based time-stamp"};
+inline static const default_tag<pmtf::scalar<float>>    TRIGGER_OFFSET = { "trigger_offset", "s", "sample delay w.r.t. the trigger (e.g.compensating analog group delays)"};
+
```

Comment:
I really like this idea, but I have some notes:
- I really like the idea of having meta data to describe the tag names.  I would pick names that match the sigmf names where possible.  I also have the question of ownership.  Should Gnuradio or Sigmf own these definitions.  (I don't know.)
- Right now, tags are kind of annoying because they aren't documented and you have to know the correct type to use it.  It would be really slick to be able to support a syntax for getting the right type using the key name.  Something like.  `auto sample_rate = tag::SAMPLE_RATE::get_from(testTag);`

Reply:
@jsallay how I understood @dkozel: this would be defined within GR, kept compatible in naming, and loosely coupled to SigMF. 

Thus the prefix `gr:` which makes the actual trigger name, for example, `gr:trigger_name` (as also given by `default_tag::key()`. There is also the `default_tag::shortKey()` method that would just yield `trigger_name`.
The actual prefix is arbitrary for now. The main goal is to have a global definition in the `gr::tag` namespace that can be shared across all blocks that choose to use these definitions.

>  It would be really slick to be able to support 

This PR already does: the assignment operator ensures that the correct PMTF type is being used for writing the meta info:
```cpp
testTag.insert_or_assign(tag::SAMPLE_RATE(4.0f));
```
If there is a type-mismatch, the above expression wouldn't even compile. 
Similarly for the get interface which is also strongly typed -- again using the PMTF definition:
```cpp
auto rate = testTag.get(tag::SAMPLE_RATE).value();
```
which returns the specified `pmtf::scalar<float>` type. 

One can of course down-cast/convert these PMTF types to other compatible types if the developer chooses to ...

Reply:
I think this is just me confusing the new variant version with the flatbuffers version.  With the variant, there isn't a need for a `pmtf::scalar<float>` wrapper, because its just as easy to get a reference to the underlying value using `std::get<float>`.  So I  believe this will need to be updated too.

Reply:
I think the variant version of the PMT will be *much* better - but we should proceed as if it doesn't exist and refactor later